lib/email_alert_signup_api.rb

```diff
@@ -20,10 +20,29 @@ def subscriber_list
   end
 
   def subscriber_list_options
-    {
-      "tags" => tags,
-      "title" => subscriber_list_title,
-    }
+    options = if facet_groups?
+                {
+                  "links" => facet_groups,
+                }
+              else
+                {
+                  "tags" => tags,
+                }
+              end
+
+    options.merge("title" => subscriber_list_title)
+  end
+
+  def facet_groups?
```

Comment:
I can only find one usage of this, line 23 - to my mind such a short method that isn't used anywhere else could be included in `subscriber_list_options`. Alternatively, rename it to `has_facet_groups?`  or similar as it's more descriptive and avoids having two methods (`facet_groups` and `facet_groups?`) that are differentiated only by a question mark and could easily be confused (that's the only bit that I'm really worried about)

Reply:
@oscarwyatt The convention in Ruby is to just add a `?` to the name of the thing you're checking if you're returning a boolean, the `has_` or `is_` is implied. 

I split out the code into to two smaller methods to make `subscriber_list_options` easy to read, so that anyone new coming to the code knows what `subscriber_list_options` is doing without having to delve too deeply.

Reply:
👍